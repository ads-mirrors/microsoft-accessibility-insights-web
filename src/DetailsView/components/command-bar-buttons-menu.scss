// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

.command-bar-buttons-menu-button {
    font-size: 16px;
    padding: 9px !important;
    color: $primary-text !important;

    &:hover {
        background-color: $neutral-2 !important;
    }

    &:focus {
        outline: 1px solid;
    }
}

.menu-list {
    top: 19px !important;
}

.menu-item {
    padding-bottom: unset !important;

    &:hover {
        background: none !important;
    }
}

.menu-button {
    align-items: flex-start !important;
    font-weight: 400 !important;
    margin-left: -8px !important;

    &:hover {
        background: 'none !important';
        color: $insights-button-hover;

        & > span {
            & > svg {
                color: $insights-button-hover;
            }
        }
    }
}

.refresh-icon {
    color: $insights-button-hover !important;
}

.command-bar-buttons-submenu {
    min-width: fit-content;

    button {
        height: 36px;
    }
}

.command-bar-buttons-menu {
    height: 17px;
    margin-left: 9px !important;
    margin-top: 8px !important;
    margin-bottom: 16px !important;
}
