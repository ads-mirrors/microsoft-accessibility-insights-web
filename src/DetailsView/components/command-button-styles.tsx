// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { makeStyles, tokens } from '@fluentui/react-components';

export const useCommandButtonStyle = makeStyles({
    assessmentButton: {
        fontWeight: '400 !important',
        paddingLeft: '5px !important',
        textDecoration: 'none !important',

        '&:focus': {
            outline: '1px solid',
        },

        '&:hover': {
            background: 'none !important',
            color: tokens.colorNeutralForeground2BrandHover,

            '& > span': {
                '& >svg': {
                    color: tokens.colorNeutralForeground2BrandHover,
                },
            },
        },
    },
});
