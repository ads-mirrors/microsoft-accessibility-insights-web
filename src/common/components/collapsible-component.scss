// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

@import '../../common/styles/common.scss';

span.collapsible-title {
    padding-left: 0.5vw;
}

.collapsible {
    cursor: pointer;
    width: auto;
    border: none;
    outline: none;
    line-height: 0.25px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
    padding-top: 5px;
    padding-bottom: 5px;
    padding-right: 27px !important;
    padding-left: 1px !important;

    h2 {
        font-weight: $normal-title-font-weight;
        font-size: 17px;
    }
}

.collapsible-content {
    overflow: hidden;
    padding-left: calc(0.5vw + 13px); // 13px is Chevron width.
}
