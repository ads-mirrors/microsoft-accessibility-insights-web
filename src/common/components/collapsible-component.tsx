// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { Button } from '@fluentui/react-components';
import { css } from '@fluentui/utilities';
import styles from 'common/components/collapsible-component.scss';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import * as React from 'react';

export interface CollapsibleComponentProps {
    header: JSX.Element;
    content: JSX.Element;
    contentClassName?: string;
    containerClassName?: string;
}

interface CollapsibleComponentState {
    showContent: boolean;
}

export class CollapsibleComponent extends React.Component<
    CollapsibleComponentProps,
    CollapsibleComponentState
> {
    constructor(props: CollapsibleComponentProps) {
        super(props);
        this.state = { showContent: true };
    }

    private onClick = (): void => {
        this.setState(prevState => ({ showContent: !prevState.showContent }));
    };

    public render(): JSX.Element {
        const showContent = this.state.showContent;
        let content: JSX.Element | null = null;

        if (showContent) {
            content = (
                <div className={css(this.props.contentClassName, styles.collapsibleContent)}>
                    {this.props.content}
                </div>
            );
        }

        return (
            <div className={css(this.props.containerClassName)}>
                <Button
                    className={styles.collapsible}
                    appearance="transparent"
                    onClick={this.onClick}
                    aria-expanded={showContent}
                >
                    {showContent ? (
                        <FluentUIV9Icon iconName="ChevronDown24Regular" />
                    ) : (
                        <FluentUIV9Icon iconName="ChevronRight24Regular" />
                    )}
                    {this.props.header}
                </Button>
                {content}
            </div>
        );
    }
}
