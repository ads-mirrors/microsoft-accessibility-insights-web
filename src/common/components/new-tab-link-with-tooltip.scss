// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../common/styles/colors.scss';

:global(.high-contrast-theme) {
    .insights-tooltip {
        outline-style: solid;
        outline-width: 1px;
        outline-color: $always-white;
    }
}

.insights-tooltip-host {
    display: inline-block;
    min-width: 0;
}

.insights-link {
    display: flex;

    &:hover,
    &:active {
        text-decoration: underline !important;
    }
}

.insights-link-icon {
    display: flex;

    &:hover,
    &:active {
        text-decoration: none !important;
    }
}

@media screen and (forced-colors: active) {
    .insights-link:hover,
    .insights-link-icon:hover {
        forced-color-adjust: none;
        color: ButtonText !important;
    }
}
