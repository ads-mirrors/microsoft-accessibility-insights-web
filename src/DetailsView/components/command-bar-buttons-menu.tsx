// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    Menu,
    MenuButton,
    MenuList,
    MenuPopover,
    MenuProps,
    MenuTrigger,
    Tooltip,
} from '@fluentui/react-components';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import { ButtonRefFunction } from 'DetailsView/components/details-view-command-bar';
import { StartOverMenuItem } from 'DetailsView/components/start-over-component-factory';
import * as React from 'react';
import styles from './command-bar-buttons-menu.scss';

export type CommandBarButtonsMenuProps = {
    renderExportReportButton: () => JSX.Element | null;
    saveAssessmentButton?: JSX.Element | null;
    loadAssessmentButton?: JSX.Element | null;
    transferToAssessmentButton?: JSX.Element | null;
    getStartOverMenuItem: () => StartOverMenuItem;
    buttonRef?: ButtonRefFunction;
    hasSubMenu?: boolean;
};

export const CommandBarButtonsMenu = NamedFC<CommandBarButtonsMenuProps>(
    'CommandBarButtonsMenu',
    props => {
        const [open, setOpen] = React.useState(false);
        const onOpenChange: MenuProps['onOpenChange'] = (e, data) => setOpen(data.open);
        const exportButton = props.renderExportReportButton();

        const overflowItems: any[] = [];
        if (exportButton != null) {
            overflowItems.push({
                key: 'export report',
                children: <>{exportButton}</>,
            });
        }
        if (props.saveAssessmentButton && props.loadAssessmentButton) {
            overflowItems.push(
                {
                    key: 'save assessment',
                    children: <>{props.saveAssessmentButton}</>,
                },
                {
                    key: 'load assessment',
                    children: <>{props.loadAssessmentButton}</>,
                },
            );
        }

        if (props.transferToAssessmentButton) {
            overflowItems.push({
                key: 'transfer to assessment',
                children: <>{props.transferToAssessmentButton}</>,
            });
        }

        overflowItems.push({
            key: 'start over',
            ...props.getStartOverMenuItem(),
        });

        return (
            <>
                <Menu open={open} onOpenChange={onOpenChange}>
                    <Tooltip content="More actions" relationship="label">
                        <MenuTrigger disableButtonEnhancement>
                            <MenuButton
                                appearance="transparent"
                                aria-label="More actions"
                                icon={<FluentUIV9Icon iconName="MoreHorizontalRegular" />}
                                className={styles.commandBarButtonsMenuButton}
                                ref={() => props.buttonRef}
                            />
                        </MenuTrigger>
                    </Tooltip>
                    <MenuPopover
                        style={{
                            border: 'unset !important',
                            borderRadius: 'unset !important',
                        }}
                    >
                        <MenuList className={styles.menuList}>
                            {overflowItems.map((item, index) => (
                                <span key={index}>{item?.children}</span>
                            ))}
                        </MenuList>
                    </MenuPopover>
                </Menu>
            </>
        );
    },
);
