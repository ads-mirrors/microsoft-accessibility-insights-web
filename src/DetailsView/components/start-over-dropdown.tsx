// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { DirectionalHint } from '@fluentui/react';

import {
    Menu,
    MenuButton,
    MenuItem,
    MenuItemProps,
    MenuList,
    MenuPopover,
    MenuTrigger,
} from '@fluentui/react-components';

import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import { ButtonRefFunction } from 'DetailsView/components/details-view-command-bar';
import { StartOverDialogType } from 'DetailsView/components/start-over-dialog';
import { useStartOverDropdownStyles } from 'DetailsView/components/start-over-dropdown-styles';
import * as React from 'react';

import { StartOverContextMenuKeyOptions } from './details-view-right-panel';

export type StartOverDropdownMenuItems = MenuItemProps & {
    name: string;
};

export interface StartOverState {
    isContextMenuVisible: boolean;
    target?: HTMLElement | string | MouseEvent | null;
}

export interface StartOverProps {
    singleTestSuffix: string;
    dropdownDirection: DropdownDirection;
    openDialog: (dialogType: StartOverDialogType) => void;
    buttonRef?: ButtonRefFunction;
    allTestSuffix: string;
    rightPanelOptions: StartOverContextMenuKeyOptions;
    switcherStartOverPreferences: StartOverContextMenuKeyOptions;
    hasSubMenu?: boolean;
    isNarrowMode?: boolean;
}

const dropdownDirections = {
    down: {
        directionalHint: DirectionalHint.bottomAutoEdge,
        iconName: 'ChevronDown',
    },
    left: {
        directionalHint: DirectionalHint.leftTopEdge,
        iconName: 'ChevronRight',
    },
};

export type DropdownDirection = keyof typeof dropdownDirections;

export const StartOverDropdown = NamedFC<StartOverProps>('StartOverDropdown', props => {
    const stylesValue: any = useStartOverDropdownStyles();
    const direction = props.dropdownDirection;
    const getMenuItemsV9 = (): StartOverDropdownMenuItems[] => {
        const {
            singleTestSuffix,
            allTestSuffix,
            rightPanelOptions,
            switcherStartOverPreferences: startOverButtonOptionPreferences,
        } = props;
        const items: StartOverDropdownMenuItems[] = [];
        const assessmentKey = {
            key: 'assessment',
            name: `Start over ${allTestSuffix}`,
            onClick: onStartOverAllTestsMenu,
        };
        const testKey = {
            key: 'test',
            name: `Start over ${singleTestSuffix}`,
            onClick: onStartOverTestMenu,
        };

        items.push(assessmentKey);

        if (rightPanelOptions.showTest && startOverButtonOptionPreferences.showTest) {
            items.push(testKey);
        }

        return items;
    };

    const onStartOverTestMenu = (): void => {
        props.openDialog('test');
    };

    const onStartOverAllTestsMenu = (): void => {
        props.openDialog('assessment');
    };
    return (
        <div>
            <Menu>
                <MenuTrigger disableButtonEnhancement>
                    <MenuButton
                        role={props.isNarrowMode ? 'menuitem' : 'button'}
                        ref={props.buttonRef}
                        appearance="transparent"
                        shape="square"
                        className={stylesValue?.menuButton}
                        icon={<FluentUIV9Icon iconName="ArrowClockwiseRegular" />}
                        aria-label="start over menu"
                        menuIcon={
                            direction === 'left' ? (
                                <FluentUIV9Icon
                                    iconName="ChevronRight20Regular"
                                    customClass={stylesValue?.chevronIcon}
                                />
                            ) : (
                                <FluentUIV9Icon
                                    iconName="ChevronDown20Regular"
                                    customClass={stylesValue?.chevronIcon}
                                />
                            )
                        }
                    >
                        Start over
                    </MenuButton>
                </MenuTrigger>
                <MenuPopover className={stylesValue?.menuPopover}>
                    <MenuList>
                        {getMenuItemsV9().map(item => (
                            <MenuItem
                                persistOnClick={false}
                                className={stylesValue?.menuItem}
                                key={item.key}
                                {...item}
                            >
                                {item.name}
                            </MenuItem>
                        ))}
                    </MenuList>
                </MenuPopover>
            </Menu>
        </div>
    );
});
