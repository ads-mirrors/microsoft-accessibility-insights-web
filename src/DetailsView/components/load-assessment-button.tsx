// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { AssessmentDataParser } from 'common/assessment-data-parser';
import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
import { FluentUIV9Icon } from 'common/icons/fluentui-v9-icons';
import { NamedFC } from 'common/react/named-fc';
import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
import { TabStoreData } from 'common/types/store-data/tab-store-data';
import { UrlParser } from 'common/url-parser';
import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
import { useCommandButtonStyle } from 'DetailsView/components/command-button-styles';
import { ButtonRefFunction } from 'DetailsView/components/details-view-command-bar';

import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
import * as React from 'react';

export type LoadAssessmentButtonDeps = {
    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
    assessmentDataParser: AssessmentDataParser;
    urlParser: UrlParser;
    loadAssessmentHelper: LoadAssessmentHelper;
};
export interface LoadAssessmentButtonProps {
    deps: LoadAssessmentButtonDeps;
    tabStoreData: TabStoreData;
    assessmentStoreData: AssessmentStoreData;
    handleLoadAssessmentButtonClick: () => void;
    isNarrowMode?: boolean;
    buttonRef?: ButtonRefFunction;
}

export const loadAssessmentButtonAutomationId = 'load-assessment-button';

export const LoadAssessmentButton = NamedFC<LoadAssessmentButtonProps>(
    'LoadAssessmentButton',
    props => {
        const loadButtonStyles = useCommandButtonStyle();
        return (
            <InsightsCommandButton
                data-automation-id={loadAssessmentButtonAutomationId}
                insightsCommandButtonIconProps={{
                    icon: <FluentUIV9Icon iconName="FolderOpenRegular" />,
                }}
                onClick={props.handleLoadAssessmentButtonClick}
                className={loadButtonStyles.assessmentButton}
                ref={props.buttonRef}
                {...props}
            >
                Load assessment
            </InsightsCommandButton>
        );
    },
);
