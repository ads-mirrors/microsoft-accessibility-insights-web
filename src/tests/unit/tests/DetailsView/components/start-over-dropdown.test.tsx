// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    Menu,
    MenuButton,
    MenuItem,
    MenuList,
    MenuPopover,
    MenuTrigger,
} from '@fluentui/react-components';
import {
    ArrowClockwiseRegular,
    ChevronRightRegular,
    ChevronDownRegular,
} from '@fluentui/react-icons';
import { fireEvent, render } from '@testing-library/react';
import { StartOverContextMenuKeyOptions } from 'DetailsView/components/details-view-right-panel';
import { StartOverDialogType } from 'DetailsView/components/start-over-dialog';
import { StartOverDropdown, StartOverProps } from 'DetailsView/components/start-over-dropdown';
import React from 'react';
import {
    expectMockedComponentPropsToMatchSnapshots,
    getMockComponentClassPropsForCall,
    mockReactComponents,
    useOriginalReactElements,
} from 'tests/unit/mock-helpers/mock-module-helpers';
import { IMock, Mock, Times } from 'typemoq';

jest.mock('@fluentui/react');
jest.mock('@fluentui/react-components');
jest.mock('@fluentui/react-icons');
jest.mock('common/components/controls/insights-command-button');
jest.mock('DetailsView/components/start-over-dropdown-styles', () => {
    return {
        useStartOverDropdownStyles: jest.fn(),
    };
});
jest.mock('common/icons/fluentui-v9-icons');

describe('StartOverDropdownTest', () => {
    mockReactComponents([
        Menu,
        MenuTrigger,
        MenuItem,
        MenuButton,
        ArrowClockwiseRegular,
        ChevronRightRegular,
        ChevronDownRegular,
        MenuPopover,
        MenuList,
    ]);
    let defaultProps: StartOverProps;
    let openDialogMock: IMock<(dialogType: StartOverDialogType) => void>;

    beforeEach(() => {
        openDialogMock = Mock.ofInstance(() => null);
        defaultProps = {
            singleTestSuffix: 'single test suffix stub',
            allTestSuffix: 'all test suffix stub',
            rightPanelOptions: { showTest: true },
            switcherStartOverPreferences: { showTest: true },
            dropdownDirection: 'down',
            openDialog: openDialogMock.object,
            buttonRef: {} as any,
        };
    });

    it('render', () => {
        const renderResult = render(<StartOverDropdown {...defaultProps} />);

        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([MenuButton]);
    });

    it('render with dropdown on left', async () => {
        const props: StartOverProps = {
            ...defaultProps,
            dropdownDirection: 'left',
        };

        const renderResult = render(<StartOverDropdown {...props} />);
        expect(renderResult.asFragment()).toMatchSnapshot();
        expectMockedComponentPropsToMatchSnapshots([MenuButton]);
    });
});

describe('StartOverDropdownTest user interaction', () => {
    let defaultProps: StartOverProps;
    let openDialogMock: IMock<(dialogType: StartOverDialogType) => void>;

    beforeEach(() => {
        openDialogMock = Mock.ofInstance(() => null);
        defaultProps = {
            singleTestSuffix: 'single test suffix stub',
            allTestSuffix: 'all test suffix stub',
            rightPanelOptions: { showTest: true },
            switcherStartOverPreferences: { showTest: true },
            dropdownDirection: 'down',
            openDialog: openDialogMock.object,
            buttonRef: {} as any,
        };
    });

    it('should open the start assessment over dialog', async () => {
        useOriginalReactElements('@fluentui/react-components', [
            'Menu',
            'MenuTrigger',
            'MenuButton',
            'MenuPopover',
            'MenuList',
            'MenuItem',
        ]);
        openDialogMock.setup(sds => sds('assessment')).verifiable(Times.once());

        const renderResult = render(<StartOverDropdown {...defaultProps} />);
        fireEvent.click(renderResult.getByRole('button'));
        getMockComponentClassPropsForCall(MenuList)
            .children.find(elem => elem.key === 'assessment')
            .props.onClick();

        openDialogMock.verifyAll();
    });

    const menuButtonOptions = [true, false];
    const optionName = 'showTest';
    const optionKey = 'test';
    menuButtonOptions.forEach(rightPanelOptionEnabled => {
        menuButtonOptions.forEach(switcherPreferencesOptionEnabled => {
            const rightPanelOptions = {
                [optionName]: rightPanelOptionEnabled,
            } as StartOverContextMenuKeyOptions;
            const switcherPreferences = {
                [optionName]: switcherPreferencesOptionEnabled,
            } as StartOverContextMenuKeyOptions;

            const shouldFindOption = rightPanelOptionEnabled && switcherPreferencesOptionEnabled;

            const casePrefix = shouldFindOption
                ? `${optionKey} item IS rendered`
                : `${optionKey} item IS NOT rendered`;

            test(`${casePrefix} - rightPanelOptions.${optionName} is ${rightPanelOptionEnabled} & switcherStartOverPreferences.${optionName} is ${switcherPreferencesOptionEnabled}`, async () => {
                defaultProps.rightPanelOptions = rightPanelOptions;
                defaultProps.switcherStartOverPreferences = switcherPreferences;

                const renderResult = render(<StartOverDropdown {...defaultProps} />);
                fireEvent.click(renderResult.getByRole('button'));
                const isStartOverOptionRendered = getMockComponentClassPropsForCall(
                    MenuList,
                ).children.some(item => item.key === optionKey);
                expect(isStartOverOptionRendered).toEqual(shouldFindOption);
            });
        });
    });

    it('should open the start test over dialog', async () => {
        openDialogMock.setup(sds => sds('test')).verifiable(Times.once());

        const renderResult = render(<StartOverDropdown {...defaultProps} />);
        fireEvent.click(renderResult.getByRole('button'));
        getMockComponentClassPropsForCall(MenuList)
            .children.find(elem => elem.key === 'test')
            .props.onClick();

        openDialogMock.verifyAll();
    });
});
