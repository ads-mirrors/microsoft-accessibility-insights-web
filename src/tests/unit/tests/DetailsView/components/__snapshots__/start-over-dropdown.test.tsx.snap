// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartOverDropdownTest render 1`] = `
<DocumentFragment>
  <div>
    <mock-menu>
      <mock-menutrigger
        disablebuttonenhancement="true"
      >
        <mock-menubutton
          appearance="transparent"
          aria-label="start over menu"
          icon="[object Object]"
          menuicon="[object Object]"
          role="button"
          shape="square"
        >
          Start over
        </mock-menubutton>
      </mock-menutrigger>
      <mock-menupopover>
        <mock-menulist>
          <mock-menuitem
            name="Start over all test suffix stub"
            persistonclick="false"
          >
            Start over all test suffix stub
          </mock-menuitem>
          <mock-menuitem
            name="Start over single test suffix stub"
            persistonclick="false"
          >
            Start over single test suffix stub
          </mock-menuitem>
        </mock-menulist>
      </mock-menupopover>
    </mock-menu>
  </div>
</DocumentFragment>
`;

exports[`StartOverDropdownTest render with dropdown on left 1`] = `
<DocumentFragment>
  <div>
    <mock-menu>
      <mock-menutrigger
        disablebuttonenhancement="true"
      >
        <mock-menubutton
          appearance="transparent"
          aria-label="start over menu"
          icon="[object Object]"
          menuicon="[object Object]"
          role="button"
          shape="square"
        >
          Start over
        </mock-menubutton>
      </mock-menutrigger>
      <mock-menupopover>
        <mock-menulist>
          <mock-menuitem
            name="Start over all test suffix stub"
            persistonclick="false"
          >
            Start over all test suffix stub
          </mock-menuitem>
          <mock-menuitem
            name="Start over single test suffix stub"
            persistonclick="false"
          >
            Start over single test suffix stub
          </mock-menuitem>
        </mock-menulist>
      </mock-menupopover>
    </mock-menu>
  </div>
</DocumentFragment>
`;

exports[`StartOverDropdownTest render with dropdown on left: MenuButton props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "start over menu",
  "children": "Start over",
  "className": undefined,
  "icon": <FluentUIV9Icon
    iconName="ArrowClockwiseRegular"
  />,
  "menuIcon": <FluentUIV9Icon
    iconName="ChevronRight20Regular"
  />,
  "role": "button",
  "shape": "square",
}
`;

exports[`StartOverDropdownTest render: MenuButton props 1`] = `
{
  "appearance": "transparent",
  "aria-label": "start over menu",
  "children": "Start over",
  "className": undefined,
  "icon": <FluentUIV9Icon
    iconName="ArrowClockwiseRegular"
  />,
  "menuIcon": <FluentUIV9Icon
    iconName="ChevronDown20Regular"
  />,
  "role": "button",
  "shape": "square",
}
`;
